diff -Naur libecap-1.0.1.org/cfgaux/config.guess libecap-1.0.1.sw/cfgaux/config.guess
--- libecap-1.0.1.org/cfgaux/config.guess	2022-03-09 03:56:18.232357080 +0000
+++ libecap-1.0.1.sw/cfgaux/config.guess	2022-03-09 05:38:33.942357080 +0000
@@ -917,6 +917,14 @@
     ppc64:Linux:*:*)
 	echo powerpc64-unknown-linux-gnu
 	exit ;;
+    sw_64:Linux:*:*)
+	case `sed -n '/^cpu model/s/^.*: \(.*\)/\1/p' < /proc/cpuinfo` in
+	  sw)   UNAME_MACHINE=sw_64 ;;
+        esac
+	objdump --private-headers /bin/sh | grep -q ld.so.1
+	if test "$?" = 0 ; then LIBC="libc1" ; else LIBC="" ; fi
+	echo ${UNAME_MACHINE}-sunway-linux-gnu${LIBC}
+	exit ;;
     alpha:Linux:*:*)
 	case `sed -n '/^cpu model/s/^.*: \(.*\)/\1/p' < /proc/cpuinfo` in
 	  EV5)   UNAME_MACHINE=alphaev5 ;;
diff -Naur libecap-1.0.1.org/cfgaux/config.sub libecap-1.0.1.sw/cfgaux/config.sub
--- libecap-1.0.1.org/cfgaux/config.sub	2022-03-09 03:56:18.232357080 +0000
+++ libecap-1.0.1.sw/cfgaux/config.sub	2022-03-09 05:37:39.462357080 +0000
@@ -242,6 +242,7 @@
 	# Some are omitted here because they have special meanings below.
 	1750a | 580 \
 	| a29k \
+	| sw_64 \
 	| alpha | alphaev[4-8] | alphaev56 | alphaev6[78] | alphapca5[67] \
 	| alpha64 | alpha64ev[4-8] | alpha64ev56 | alpha64ev6[78] | alpha64pca5[67] \
 	| am33_2.0 \
@@ -322,6 +323,7 @@
 	# Recognize the basic CPU types with company name.
 	580-* \
 	| a29k-* \
+	| sw_64-* \
 	| alpha-* | alphaev[4-8]-* | alphaev56-* | alphaev6[78]-* \
 	| alpha64-* | alpha64ev[4-8]-* | alpha64ev56-* | alpha64ev6[78]-* \
 	| alphapca5[67]-* | alpha64pca5[67]-* | arc-* \
